src/librustc_trans/mir/statement.rs

```diff
@@ -27,7 +27,7 @@ impl<'a, 'tcx> MirContext<'a, 'tcx> {
         debug!("trans_statement(statement={:?})", statement);
 
         let (scope, span) = self.debug_loc(statement.source_info);
-        debuginfo::set_source_location(self, &bcx, scope, span);
+        debuginfo::set_source_location(&self.debug_context, &bcx, scope, span);
```

Comment:
Maybe this pair should be combined into `self.set_debug_loc(&bcx, statement.source_info);`?